### PR TITLE
feat(pr-review): context-aware review with change classification

### PR DESCRIPTION
**Pull request by koydas**
Introduces an explicit classification phase before generating review
findings so that documentation-only, CI/CD-only, configuration-only,
and dependency-update PRs no longer receive irrelevant test-coverage
complaints.

New flow:
  PR → Change Classification → Review Criteria Selection → Findings

Changes:
- scripts/lib/change_classifier.mjs: new module that inspects changed
  file paths (reusing extractChangedFiles from coverage_checker) and
  produces a structured context block with change_type,
  detected_categories, has_executable_code_changes, tests_expected, and
  tests_expected_reason.  Automation-scope .md files (prompts/,
  docs/code-generation.md) are correctly treated as behavioural rather
  than documentation.
- scripts/tests/change_classifier.test.mjs: 29 unit tests covering
  classifyChangedFiles, determineTestExpectation, and
  buildChangeClassificationContext across all change types and edge
  cases.
- prompts/pr-review-system.md: added "Context-aware review" rule block
  instructing the LLM to read the classification context, honour
  tests_expected, and apply criteria appropriate to the detected type
  (documentation, ci_cd, configuration, dependency_update, test_only,
  feature/bugfix/refactor/security, mixed).
- prompts/pr-review-user.md: added "Change Classification" section to
  the review output template; raised word limit to 250 to accommodate
  the new section; added explicit constraint to omit test-coverage
  findings when tests_expected is false.
- scripts/pr_review.mjs: import buildChangeClassificationContext and
  append it to the user prompt alongside the existing automation gate
  context.

https://claude.ai/code/session_01Kv5EHWB7ZhzjeP8NK3j7rp

### CODE REVIEW

**Comment by koydas** [comment]
## 🔍 Automated Code Review

APPROVED

The diff introduces a context-aware PR review system with proper test coverage for the new logic, updated documentation, and correct prompt instructions. All three automation gate requirements are satisfied:

(a) Unit-test status: ✅ 29 unit tests added in `scripts/tests/change_classifier.test.mjs` cover all classification edge cases  
(b) Documentation: ✅ `docs/code-generation.md` explains the new classification logic, patterns, and expectations  
(c) Coverage policy: ✅ CI already enforces coverage for automation logic via `scripts/lib/checkpoint.mjs`, and new classifier logic is appropriately tested  

No unintended regressions or security issues detected. The file classification patterns align with the design document, and tests_expected logic is correctly implemented in both the classifier and review prompts.

File: scripts/lib/change_classifier.mjs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    if (catSet.has('dependency_update') && catSet.size === 1) {
+      return {
+        tests_expected: false,
+        reason: 'Dependency version update; tests expected only if runtime behavior changes.',
+      };
+    }
+    // ci_cd, configuration, or any other non-behavioral mix
+    return {
+      tests_expected: false,
+      reason: 'Only non-behavioral files changed (documentation, CI/CD, configuration, or dependencies).',
```
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/lib/change_classifier.mjs:157]: Fixed in `24e6cab` — `.github/workflows/*` files are matched by `isAutomationScopeFile`, so the `isCiCdFile` branch now also sets `hasCode = true` for those paths. A workflow-only PR produces `tests_expected: true`, preserving the existing automation-gate test requirement. The corresponding test was updated accordingly.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/lib/change_classifier.mjs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  /^\.prettierrc/,
+  /^tsconfig.*\.json$/,
+  /^vitest\.config\./,
+  /^jest\.config\./,
+  /^babel\.config\./,
+];
+
+const DEPENDENCY_PATTERNS = [
+  /^package\.json$/,
+  /^package-lock\.json$/,
```
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/lib/change_classifier.mjs:35]: Fixed in `24e6cab` — `package.json` has been removed from `DEPENDENCY_PATTERNS`. It now falls through to `hasCode = true`, producing `tests_expected: true`. Only auto-generated lock files (`package-lock.json`, `yarn.lock`, `pnpm-lock.yaml`) remain in the dependency category. The comment in the pattern table explains the reason.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/pr_review.mjs [hunk 3]
```diff
@@ -168,7 +169,7 @@ const prBody = prMeta.body || '(no description provided)';
 const diff = filterDiff(rawDiff);
 
 const baseUserPrompt = interpolatePrompt(userPromptTemplate, { diff, issueTitle: prTitle, issueBody: prBody });
-const userPrompt = `${baseUserPrompt}${buildAutomationGateContext(rawDiff)}`;
+const userPrompt = `${baseUserPrompt}${buildChangeClassificationContext(rawDiff)}${buildAutomationGateContext(rawDiff)}`;
```
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/pr_review.mjs:172]: Fixed in `24e6cab` — added a "PR Review: Context-Aware Change Classification" section to `docs/code-generation.md` documenting the classification rules, `tests_expected` semantics, the `package.json`/lock-file distinction, and how the two context blocks (classification + automation gate) interact. `CHANGELOG.md` also updated as required by the changelog gate.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by koydas** [comment]
@Codex review

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/lib/change_classifier.mjs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+  const changedFiles = extractChangedFiles(rawDiff);
+  if (!changedFiles.length) return '';
+
+  const { categories, hasCode } = classifyChangedFiles(changedFiles);
+  const { tests_expected, reason } = determineTestExpectation(categories, hasCode);
+
+  const classification = categories.length > 0 ? categories.join(', ') : 'none';
+  const changeType = hasCode
+    ? categories.length > 0 ? 'mixed' : 'feature_or_bugfix'
+    : categories.length === 1 ? categories[0] : categories.length > 1 ? 'mixed' : 'unknown';
```
[the remark this answers is not in the document]
Comment (koydas) [reply scripts/lib/change_classifier.mjs]: Fixed in `b153295` — introduit `hasUncategorizedCode` dans `classifyChangedFiles` (true uniquement pour les fichiers qui ne matchent aucune table de patterns). La formule de `changeType` l'utilise : un PR workflow-only (`categories=['ci_cd'], hasUncategorizedCode=false`) retourne `change_type: ci_cd` ; un PR code+docs (`categories=['documentation'], hasUncategorizedCode=true`) retourne `change_type: mixed`. `hasCode` conserve ses sémantiques pour `determineTestExpectation`.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by koydas** [comment]
@Codex review

---
_Generated by [Claude Code](https://claude.ai/code)_
